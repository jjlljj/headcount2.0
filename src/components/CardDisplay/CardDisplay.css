.card-display {
  display: flex;
  flex-wrap: wrap;
  margin-top: 20px;
  justify-content: center;
}