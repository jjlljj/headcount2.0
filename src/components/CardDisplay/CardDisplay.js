import React, { Component } from 'react';
import Card from '../Card/Card'
import './CardDisplay.css';

const CardDisplay = ({ cards }) => {

  const cardKeys = Object.keys(cards)

  const renderedCards = cardKeys.map( key => ( <Card key={cards[key].location} card={cards[key]} /> ))

  return (
    <div className="card-display"> 
      { renderedCards }
    </div>
  );

}

export default CardDisplay;
