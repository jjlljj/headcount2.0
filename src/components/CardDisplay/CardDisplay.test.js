/* eslint-disable */ 

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import CardDisplay from './CardDisplay';

describe('CardDisplay', () => {

  it('should exist', () => {
    const mockData = { "Denver, Co": { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } }
    }

    const renderedComponent = shallow(<CardDisplay cards={ mockData } />)

    expect(renderedComponent).toBeDefined()
  })

  it('should render child cards', ()=> {
    const mockData = { "Denver, Co": { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } },
      "Boston, Ma": { location: 'Boston, Ma', data: { 2004: 0.023, 2005: 3.112, 2006: 7.110 } }
    }
    const renderedComponent = shallow(<CardDisplay cards={ mockData }/>)

    expect(renderedComponent.find('Card').length).toEqual(2)
   })

  it('should match snapshot', ()=> {
    const mockData = { "Denver, Co": { location: 'Denver, Co', data: { 2004: 0.007, 2005: 1.112, 200: 7.110 } },
      "Boston, Ma": { location: 'Boston, Ma', data: { 2004: 0.023, 2005: 3.112, 2006: 7.110 } }
    }
    const renderedComponent = shallow(<CardDisplay cards={ mockData }/>)

     expect(renderedComponent).toMatchSnapshot();
  })

});