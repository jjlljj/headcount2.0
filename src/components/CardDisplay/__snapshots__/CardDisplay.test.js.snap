// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDisplay should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardDisplay
    cards={
        Object {
            "Boston, Ma": Object {
              "data": Object {
                "2004": 0.023,
                "2005": 3.112,
                "2006": 7.11,
              },
              "location": "Boston, Ma",
            },
            "Denver, Co": Object {
              "data": Object {
                "200": 7.11,
                "2004": 0.007,
                "2005": 1.112,
              },
              "location": "Denver, Co",
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          card={
                    Object {
                              "data": Object {
                                "200": 7.11,
                                "2004": 0.007,
                                "2005": 1.112,
                              },
                              "location": "Denver, Co",
                            }
          }
          clickHandler={undefined}
/>,
        <Card
          card={
                    Object {
                              "data": Object {
                                "2004": 0.023,
                                "2005": 3.112,
                                "2006": 7.11,
                              },
                              "location": "Boston, Ma",
                            }
          }
          clickHandler={undefined}
/>,
      ],
      "className": "card-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Denver, Co",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "data": Object {
              "200": 7.11,
              "2004": 0.007,
              "2005": 1.112,
            },
            "location": "Denver, Co",
          },
          "clickHandler": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Boston, Ma",
        "nodeType": "function",
        "props": Object {
          "card": Object {
            "data": Object {
              "2004": 0.023,
              "2005": 3.112,
              "2006": 7.11,
            },
            "location": "Boston, Ma",
          },
          "clickHandler": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            card={
                        Object {
                                    "data": Object {
                                      "200": 7.11,
                                      "2004": 0.007,
                                      "2005": 1.112,
                                    },
                                    "location": "Denver, Co",
                                  }
            }
            clickHandler={undefined}
/>,
          <Card
            card={
                        Object {
                                    "data": Object {
                                      "2004": 0.023,
                                      "2005": 3.112,
                                      "2006": 7.11,
                                    },
                                    "location": "Boston, Ma",
                                  }
            }
            clickHandler={undefined}
/>,
        ],
        "className": "card-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Denver, Co",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "data": Object {
                "200": 7.11,
                "2004": 0.007,
                "2005": 1.112,
              },
              "location": "Denver, Co",
            },
            "clickHandler": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Boston, Ma",
          "nodeType": "function",
          "props": Object {
            "card": Object {
              "data": Object {
                "2004": 0.023,
                "2005": 3.112,
                "2006": 7.11,
              },
              "location": "Boston, Ma",
            },
            "clickHandler": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
